### DISCUSSION TITLE
How to reduce the number of topics when running a ros 2 application

### DISCUSSION THREAD

**Discussion by Syamkamalnadhc** [eProsima/Fast-DDS#5296] [Q&A]
I run a simple single publisher ROS2-Humble application.  
I Enabled FastDDS info logs and observed the below topics (other than the topic given in application for communication). 
1. ros_discovery_info
2. rt/rosout
3. rr/publisher_node/get_parametersReply
4. rr/publisher_node/set_parametersReply
5. rr/publisher_node/set_parameters_atomicallyReply
6. rr/publisher_node/describe_parametersReply
7. rr/publisher_node/list_parametersReply
8. rt/parameter_events
9. rq/subscriber_node/get_parametersRequest
10. rq/publisher_node/set_parametersRequest
11. rq/subscriber_node/get_parameter_typesRequest
12. rq/publisher_node/set_parameters_atomicallyRequest
13. rq/publisher_node/describe_parametersRequest
14. rq/publisher_node/list_parametersRequest
15. rr/subscriber_node/get_parameter_typesReply
16. rq/subscriber_node/set_parametersRequest
17. rr/subscriber_node/set_parametersReply
18. rq/subscriber_node/set_parameters_atomicallyRequest
19. rr/subscriber_node/set_parameters_atomicallyReply
20. rq/subscriber_node/describe_parametersRequest
21. rr/subscriber_node/describe_parametersReply
22. rq/subscriber_node/list_parametersRequest
23. rr/subscriber_node/list_parametersRepl

Queries

1. Are these topics necessary for a communication in ROS2?
2. How to run a ROS2 application with minimum number of topics?
3. What is the relevance of these topics in ROS2?

**Comment by JesusPoderoso** [comment]
Hi @Syamkamalnadhc, thanks for using Fast DDS.
Those topics that contains the `rr/` (ROS 2 Service Response) and `rq/` (ROS 2 Service Request) prefixes, and the `Reply` and `Request` suffixes (respectively), in the topic names are part of the `rt/parameter_events` service, launched by default in ROS 2.

> 1. Are these topics necessary for a communication in ROS2?

They are not necessary, but useful. Check their usage in this [ROS 2 documentation section](https://docs.ros.org/en/rolling/How-To-Guides/Node-arguments.html#parameters).

> 2. How to run a ROS2 application with minimum number of topics?

You should include the following C++ snippet in your application code.
``` cpp
rclcpp::NodeOptions options;
options.start_parameter_services(false);
options.start_parameter_event_publisher(false);
Node("example_node", options);
```

> 3. What is the relevance of these topics in ROS2?

Please, refer to the [ROS 2 parameters documentation](https://docs.ros.org/en/rolling/Concepts/Basic/About-Parameters.html).
